src/Command.hs

```diff
@@ -76,7 +86,7 @@ cli = info (helper <*> parser) $ mempty
 
 -- | Run a given command
 run :: CLI -> IO ()
-run = handle prettyIOException . \case
+run = withUtf8Encoding . handle prettyIOException . \case
```

Comment:
I read the test case which failed... the text decode failure happens during option parsing.
This is setting the encoding too late. The option parsing is already done.

Reply:
Good point.